_analyzers/stemming.md

```diff
@@ -0,0 +1,163 @@
+      "type": "<ALPHANUM>",
+      "position": 3
+    },
+    {
+      "token": "swiftli",
+      "start_offset": 24,
+      "end_offset": 31,
+      "type": "<ALPHANUM>",
+      "position": 4
+    }
+  ]
+}
+```
+
+## Stemmers categories
+
+There are two categories of stemmers that can be configured:
+
+- [Algorithmic stemmers]({{site.url}}{{site.baseurl}}/analyzers/stemming/#algorithmic-stemmers)
+- [Dictionary stemmers]({{site.url}}{{site.baseurl}}/analyzers/stemming/#dictionary-stemmers)
+
+### Algorithmic stemmers
+
+Algorithmic stemmers apply predefined rules to systematically strip affixes (prefixes and suffixes) from words, reducing them to their stems. See following list of token filters examples that use algorithmic stemmers:
+
+[`porter_stem`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/porter-stem/): Applies the Porter stemming algorithm, primarily for English, to remove common suffixes and reduce words to their stems. For example, "running" becomes "run".
+
+[`kstem`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/kstem/): A lightweight stemmer designed for English that combines algorithmic stemming with a built-in dictionary. It reduces plurals to singulars, converts verb tenses to their base forms, and removes common derivational endings. 
+
+
+[`stemmer`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/stemmer/): Provides algorithmic stemming for various languages, including English, with options for different stemming algorithms like `light_english`, `minimal_english`, and `porter2`. 
+
+
+[`snowball`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/snowball/): Utilizes the Snowball algorithm to provide efficient and accurate stemming for multiple languages, including English, French, German, and others. 
+
+### Dictionary stemmers
+
+Dictionary stemmers rely on extensive dictionaries to map words to their root forms, effectively handling irregular words. They look up each word in a precompiled list to find its corresponding stem. This operation is more resource intensive but often yields better results for irregular words and words which might appear to have a similar stem but are very different in their meaning.
+
+The most prominent example of a dictionary stemmer is [`hunspell`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/hunspell/) token filter, which leverages Hunspell, a spell checker engine used in many open-source applications.
+
+### Consideration
+
+- Algorithmic Stemmers are suitable when processing speed and memory efficiency are priorities, and the language has relatively regular morphological patterns.
+- Dictionary Stemmers are ideal when accuracy in handling irregular word forms is crucial, and resources are available to support the increased memory usage and processing time.
+
+Choose algorithmic stemmers for speed and efficiency with regular languages. Opt for dictionary stemmers for accuracy with complex languages, acknowledging the increased resource requirements.
```

Comment:
Can we be more specific and replace "regular" with a more descriptive adjective? How about:
```suggestion
Choose algorithmic stemmers for speed and efficiency when processing words in languages with consistent morphological patterns. Use dictionary stemmers for greater accuracy in complex languages, provided that the increased resource requirements are acceptable.
```